modules/loader/src/test/scala/com/snowplowanalytics/snowplow/rdbloader/experimental/LoaderSpecification.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2012-2023 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and
+ * limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.experimental
+
+import scala.concurrent.duration.DurationInt
+
+import doobie.ConnectionIO
+
+import cats.effect.{IO, Resource}
+import cats.effect.std.Dispatcher
+import cats.effect.unsafe.implicits.global
+
+import fs2.concurrent.SignallingRef
+
+import org.http4s.blaze.client.BlazeClientBuilder
+
+import com.snowplowanalytics.snowplow.rdbloader.config.{Config, StorageTarget}
+import com.snowplowanalytics.snowplow.rdbloader.dsl.metrics.Metrics
+import com.snowplowanalytics.snowplow.rdbloader.dsl.{Logging, Monitoring, Transaction}
+import com.snowplowanalytics.snowplow.rdbloader.common.LoaderMessage
+import com.snowplowanalytics.snowplow.rdbloader.common.cloud.{Queue, SecretStore}
+import com.snowplowanalytics.snowplow.rdbloader.common.integrationtestutils.ItUtils._
+import com.snowplowanalytics.snowplow.rdbloader.common.integrationtestutils.InputBatch
+
+import org.specs2.mutable.Specification
+
+abstract class LoaderSpecification extends Specification with TestDAO.Provider with StorageTargetProvider with AzureTestResources {
+  skipAllIf(anyEnvironmentVariableMissing())
+  import LoaderSpecification._
+
+  def run[A](
+    inputBatches: List[InputBatch],
+    countExpectations: CountExpectations,
+    dbActions: TestDAO => IO[A]
```

Comment:
As it's mostly about quering stuff from DB... can we name it that way? Actions sounds pretty generic ;) Something like: `queryDbOutput`? And instead of just `A` type param I would probably go with `DB_OUTPUT`. WDYT?